src/pykeen/datasets/__init__.py

```diff
@@ -189,6 +192,45 @@ def get_dataset(
     )
 
 
+def _digest_kwargs(dataset_kwargs: Mapping[str, Any]) -> str:
+    digester = hashlib.sha256()
+    for key in sorted(dataset_kwargs.keys()):
+        digester.update(key.encode(encoding="utf8"))
+        digester.update(str(dataset_kwargs[key]).encode(encoding="utf8"))
+    return base64.urlsafe_b64encode(digester.digest()).decode("utf8")[:32]
```

Comment:
why not `digester.hexdigest()[:32]`?

Reply:
Because `urlsafe` uses a larger alphabet, and hence can represent more unique values in 32 characters -> smaller probability of collision.